### PR TITLE
Optimize term-timeout wait.

### PR DESCRIPTION
**Pull request by kurtisvg**
Add a small optimization to better cap the max amount of time the proxy spends waiting for incoming connections to close. 

/cc @Carrotman42

### CODE REVIEW

File: proxy/proxy/client.go [hunk 1]
```diff
@@ -359,9 +359,16 @@ func NewConnSrc(instance string, l net.Listener) <-chan Conn {
 // close. Returns an error if there are still active connections after waiting
 // for the whole length of the timeout.
 func (c *Client) Shutdown(termTimeout time.Duration) error {
-	termTime := time.Now().Add(termTimeout)
-	for termTime.After(time.Now()) && atomic.LoadUint64(&c.ConnectionsCounter) > 0 {
-		time.Sleep(time.Second)
+	term, tick := time.After(termTimeout), time.Tick(100 * time.Millisecond)
+	for {
+		select {
+			case <- tick:
```
Comment (Carrotman42) [inline proxy/proxy/client.go:365]: Please run gofmt.

File: proxy/proxy/client.go [hunk 2]
```diff
@@ -359,9 +359,16 @@ func NewConnSrc(instance string, l net.Listener) <-chan Conn {
 // close. Returns an error if there are still active connections after waiting
 // for the whole length of the timeout.
 func (c *Client) Shutdown(termTimeout time.Duration) error {
-	termTime := time.Now().Add(termTimeout)
-	for termTime.After(time.Now()) && atomic.LoadUint64(&c.ConnectionsCounter) > 0 {
-		time.Sleep(time.Second)
+	term, tick := time.After(termTimeout), time.Tick(100 * time.Millisecond)
```
Comment (Carrotman42) [inline proxy/proxy/client.go:362]: nit: `defer tick.Stop()`. Even though this is (currently) on only the shutdown path, it seems prudent to not leak resources anyway.

Comment (kurtisvg) [reply proxy/proxy/client.go:362]: `time.Tick` returns a channel and can't be stopped, but I can use `NewTicker` instead.

Comment (Carrotman42) [reply proxy/proxy/client.go:362]: Whoops duh, thanks for figuring out what I meant.

**Pull request merged**